### ISSUE TITLE
Bring herald/db/prefixes.go up to date with new tables

### ISSUE THREAD

**Issue by moodyjon** [lbryio/herald.go#46]
Latest tables in (python) hub:

https://github.com/lbryio/hub/blob/4845b9224842e6800808aa1860330b34929b06ae/hub/db/common.py#L50

```
    channel_count = b'Z'
    support_amount = b'a'
    block_tx = b'b'
    trending_notifications = b'c'
    mempool_tx = b'd'
    touched_hashX = b'e'
    hashX_status = b'f'
    hashX_mempool_status = b'g'
```

Latest tables in herald.go:

https://github.com/lbryio/herald.go/blob/8b4b27bdb95ed899a055a9c0ebd8f10e7ca4b5e6/db/prefixes/prefixes.go#L61

```
	ChannelCount  = 'Z'
	SupportAmount = 'a'
	BlockTXs      = 'b'
```

**Comment by moodyjon** [comment]
After writing some Pack/Unpack() code for one of the new tables, I'm searching for something that can automate this. Using reflect to make the Pack/Unpack() generic is not a new idea. Found these packages that do it:

https://pkg.go.dev/github.com/lunixbochs/struc
https://pkg.go.dev/github.com/go-restruct/restruct

Performance benchmark from 7 years ago indicated it was 10x slower than manually written code.
https://github.com/lunixbochs/struc#benchmark

**Comment by lothropgauss** [comment]
> After writing some Pack/Unpack() code for one of the new tables, I'm searching for something that can automate this. Using reflect to make the Pack/Unpack() generic is not a new idea. Found these packages that do it:
> 
> https://pkg.go.dev/github.com/lunixbochs/struc https://pkg.go.dev/github.com/go-restruct/restruct
> 
> Performance benchmark from 7 years ago indicated it was 10x slower than manually written code. https://github.com/lunixbochs/struc#benchmark

The plan is to use generics from go 1.18 for this https://go.dev/blog/why-generics. If we can get performance improvements out of this tangentially to that, it sounds good. We should talk about this in the call.

**Comment by moodyjon** [comment]
I have successfully tagged 2 existing structs (EffectiveAmountKey/Value) structs and ran Pack()/Unpack() based on https://pkg.go.dev/github.com/go-restruct/restruct and reflection.

I also set up a testing framework where the different Pack/Unpack() implementations can be tested against each other. For example, manual Pack() with "restruct" Unpack(), and vice versa. 

TODO: Need to gather some test data for the new tables and finish the manually-implemented Pack/Unpack() for them.
